### ISSUE TITLE
Hooks (part 1)

### ISSUE THREAD

**Issue by lilydjwg** [archlinuxcn/repo#233]
See [here](https://www.archlinux.org/todo/hooks-part-1/) for more info.

**Comment by zsrkmyn** [comment]
There is a rough way to find which packages need to be fixed by using the one-line script:

`cd repo; find -name *.install -exec grep '\(update-desktop-database\|update-mime-database\|install-info\|glib-compile-schemes\|gtk-update-icon-cache\|xdg-icon-resource\|gconf\|gio-querymodules\)' {} +|awk -F'/' '{print $2}'|uniq`

And I find these packages:

- [ ] spotify
- [ ] gitter
- [ ] rubymine
- [ ] xfce4-linelight-plugin
- [ ] sublime-text-imfix
- [ ] foxitreader
- [ ] zeal-git
- [ ] xfce4-pulseaudio-plugin
- [ ] gnome-panel2
- [ ] kreogist-mu
- [ ] global
- [ ] lib32-libindicator
- [ ] google-chrome
- [ ] mendeleydesktop
- [ ] xfce4-soundmenu-plugin
- [ ] rstudio-desktop-bin
- [ ] vim-lily
- [ ] youtube-dl-gui-git
- [ ] nautilus-dropbox
- [ ] lib32-gconf
- [ ] musescore-git
- [ ] drive
- [ ] xfdashboard
- [ ] google-chrome-dev
- [ ] inkscape-gtk3-bzr
- [ ] thunar-dropbox
- [ ] icecat
- [ ] lantern
- [ ] pystopwatch
- [ ] xware-desktop
- [ ] vuze
- [ ] xfce4-kbdleds-plugin
- [ ] plasmate
- [ ] vokoscreen
- [ ] pointdownload-git
- [ ] gottet
- [ ] qtadb
- [ ] firefox-kde-opensuse
- [ ] telegram-desktop
- [ ] numix-circle-icon-theme-git
- [ ] zeal
- [ ] eaglemode
- [ ] chromium-dev
- [ ] xfce4-windowck-plugin
- [ ] libxfce4ui-devel
- [ ] wxmedit
- [ ] xfce4-indicator-plugin
- [ ] google-earth
- [ ] numix-icon-theme-git
- [ ] iksemel
- [ ] shadowsocks-qt5
- [ ] acroread
- [ ] megasync
- [ ] flashplugin-10
- [ ] pnmixer
- [ ] vicare-scheme-git
- [ ] gnome-search-tool-no-nautilus
- [ ] xpra-winswitch
- [ ] sublime-text-dev
- [ ] freecad
- [ ] youdao-dict
- [ ] vidalia
- [ ] julia-git
- [ ] vokoscreen-git
- [ ] kdeconnect-git
- [ ] pasystray
- [ ] masterpdfeditor
- [ ] inkscape-bzr
- [ ] jdk7
- [ ] wget-lily
- [ ] gtk-theme-config
- [ ] atom-editor
- [ ] fcitx-lilydjwg-git
- [ ] wallch
- [ ] xfdesktop-devel
- [ ] sublime-text-dev-imfix
- [ ] htmldoc
- [ ] gcc-multilib-x32
- [ ] go-for-it-git
- [ ] wps-office
- [ ] pcmanfm-qt-git
- [ ] byzanz-git
- [ ] jdk
- [ ] wireshark-gtk2
- [ ] leo
- [ ] sublime-text
- [ ] gcc-gcj
- [ ] android-studio
- [ ] qpdfview
- [ ] libgweather2
- [ ] balsamiqmockups
- [ ] corebird-git

I'm not sure whether the list is correct or not, for I'm not familiar with pacman hooks. Any one could help edit this comment if there is anything wrong. Thanks in advance.

Related articles:
- man 5 alpm-hooks
- [DeveloperWiki:Pacman_Hooks](https://wiki.archlinux.org/index.php/DeveloperWiki:Pacman_Hooks)

**Comment by colinkeenan** [comment]
Is it enough to just rebuild our packages, or do we have to wait for someone to change the PKGBUILD first?

**Comment by yuyichao** [comment]
IIUC you should delete the unnecessary install script, update pkgrel and rebuild.

**Comment by colinkeenan** [comment]
Or, just delete those lines from the install script? If deleting the
install script, I would also need to remove `install="${pkgname}.install"`
from the PKGBUILD.

**Comment by farseerfc** [comment]
@colinkeenan We need to remove the calls to update-desktop-database/gtk-update-icon-cache/... from the $pkgname.install file (and remove the $pkgname.install file also if nothing is left), but remember to KEEP the dependencies (i.e. leave depends array untouched).

My opinion is that if the PKGBUILD came from AUR and still tracks AUR, we can wait for AUR maintainer to do the change and rebuild accordingly. The only drawback of not rebuilding is to run those commands twice, once in the .install script and once in the hooks, so not a severe issue.

**Comment by xgdgsc** [comment]
When I try to rm the .install file from AUR, at `git push` I get an error 

```
remote: error: missing source file: mendeleydesktop.install
remote: error: hook declined to update refs/heads/master
To ssh://aur@aur4.archlinux.org/mendeleydesktop.git
 ! [remote rejected] master -> master (hook declined)
```

**Comment by farseerfc** [comment]
Remember to run: `makepkg --printsrcinfo >.SRCINFO` every time you create a commit for AUR.
You may need to use `git commit --amend` now to rewrite history.

**Comment by colinkeenan** [comment]
you can just use `mksrcinfo`

**Comment by xgdgsc** [comment]
So the mksrcinfo must be in the same commit as the `git rm`? I did that in a later commit. I will try a gain.

**Comment by farseerfc** [comment]
Yes they must in the same commit. You need to rewrite the history.

**Comment by lilydjwg** [comment]
好多还没有处理的包啊。

**Comment by petronny** [comment]
这些命令被删除是因为被弃用了吗？

**Comment by yuyichao** [comment]
因为会自动检测运行.

**Comment by zsrkmyn** [comment]
@petronny, 因为 alpm-hooks 在 `pacman` 更新/删除/安装软件包的时候会检测文件或软件包变动，然后执行相关指令。好处在于多个包对于一个 hook 的话不会重复执行。比如安装了多个 ttf-* 字体，只会在安装完毕后执行一次`/usr/bin/fc-cache -s`，减少重复的工作以及简化`.install`脚本。
看[这里](https://github.com/archlinuxcn/repo/issues/233#issuecomment-216759604)的 Related articles 啦。

**Comment by petronny** [comment]
原来如此。。总之我先通知AUR的维护者了

**Comment by lilydjwg** [comment]
这里还有多少包没有完成呀？

**Comment by petronny** [comment]
我对现在的repo重跑了一遍@zsrkmyn的命令，并更新了一下list

**Comment by lilydjwg** [comment]
我重新生成了一个新 list，放到第一个评论里了。
还有好多包……

**Comment by petronny** [comment]
更新了list

**Comment by lilydjwg** [comment]
list updated again.

**Comment by SilverRainZ** [comment]
list updated again again.

**Comment by oldherl** [comment]
Command update: (Use maxdepth to limit depth to prevent finding files in the src/ directory that are not actually installed)
```
cd repo/archlinuxcn; find -maxdepth 2 -name '*.install' -exec grep '\(update-desktop-database\|update-mime-database\|install-info\|glib-compile-schemes\|gtk-update-icon-cache\|xdg-icon-resource\|gconf\|gio-querymodules\)' {} +|cut -d'/' -f2|sort -u|sed 's/^/* [ ] /'
```
List update:
* [ ] eka2l1-git
* [ ] emacs-org-mode-git
* [ ] gconf
* [ ] gdb-git
* [ ] go-for-it-git
* [ ] jdownloader2
* [ ] konsole-git
* [ ] logisim
* [ ] mars-mips
* [ ] menulibre
* [ ] notion-app-enhanced
* [ ] osdlyrics-git
* [ ] panoply
* [ ] pnmixer
* [ ] pystopwatch
* [ ] rstudio-desktop-bin
* [ ] snes9x-git
* [ ] vlc-nox

**Comment by lilydjwg** [comment]
这都谁的包啊，[at-maintainer](https://github.com/archlinuxcn/lilac/blob/master/scripts/at-maintainer) 一下？

**Comment by oldherl** [comment]
> 这都谁的包啊，[at-maintainer](https://github.com/archlinuxcn/lilac/blob/master/scripts/at-maintainer) 一下？

等两天后的那一大波丢包之后我再更新一下吧……

**Comment by Integral-Tech** [comment]
Fixed it https://github.com/archlinuxcn/repo/commit/ec167f1689d85fad7c9ecd4abb06c25635f6e3e3
